movemeant/push_notifications/views.py

```diff
@@ -7,20 +7,6 @@
 from .serializers import DeviceSerializer
 
 
-class DeviceCreateView(APIView):
-    permission_classes = (IsAuthenticated, )
-
-    def post(self, request, format=None):
-        user = request.user
-        device_token = request.data.get('device', None)
-        
-        if device_token is None:
-            return Response(status=status.HTTP_400_BAD)
-
-        
-        obj, created = Device.objects.get_or_create(
-            user=user,
-            token=device_token
-        )
-
-        return Response(status=status.HTTP_200_OK)
\ No newline at end of file
+class DeviceCreateView(CreateAPIView):
```

Comment:
if possible let's use the generic APIView version of this endpoint as that is how I configured the apps to register their tokens. Otherwise I need to submit a whole object of data (the Device Model Serializer) which contains info the app may not necessarily know on cold start.

Reply:
done!